src/overlays/actors/ovl_En_Tanron3/z_en_tanron3.h

```diff
@@ -7,10 +7,53 @@ struct EnTanron3;
 
 typedef void (*EnTanron3ActionFunc)(struct EnTanron3*, GlobalContext*);
 
+typedef enum {
+    /* 0 */ TIMER_PICK_DIRECTION_OR_DIE,
+    /* 1 */ TIMER_OUT_OF_WATER,
+    /* 2 */ TIMER_ATTACK_OR_WAIT,
+    /* 3 */ TIMER_MAX
+} TimerWork;
+
+typedef struct {
+    /* 0x00 */ u8 unk_00;
+    /* 0x02 */ s16 unk_02;
+    /* 0x04 */ Vec3f unk_04;
+    /* 0x10 */ Vec3f unk_10;
+    /* 0x1C */ Vec3f unk_1C;
+    /* 0x28 */ char unk_28[0xC];
+    /* 0x34 */ Vec3f unk_34;
+    /* 0x40 */ char unk_40[0x4];
+} UnkTanron3Effect;
```

Comment:
You can probably work out what this does by looking at other things that use this system, particularly in OoT.

Reply:
Nothing in MM currently uses `globalCtx->specialEffects`. In OoT, bosses frequently use this, but they define things in quite different ways. I documented what I could, but I think the full documentation is going to have to wait until Gyorg itself is decompiled; I suspect this is *actually* a Gyorg effect struct that just so happens to be used by Tanron3 too.